instrumentation/lettuce/lettuce-5.2/src/main/java8/io/opentelemetry/auto/instrumentation/lettuce/v5_2/OpenTelemetryTracing.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    public Context getContext() {
+      return context;
+    }
+  }
+
+  static class OpenTelemetryEndpoint implements Endpoint {
+    final String ip;
+    final int port;
+    @Nullable final String name;
+
+    protected OpenTelemetryEndpoint(String ip, int port, @Nullable String name) {
+      this.ip = ip;
+      this.port = port;
+      if (!ip.equals(name)) {
+        this.name = name;
+      } else {
+        this.name = null;
+      }
+    }
+  }
+
+  static class OpenTelemetryTracer extends Tracer {
+
+    protected OpenTelemetryTracer() {}
+
+    @Override
+    public OpenTelemetrySpan nextSpan() {
+      return new OpenTelemetrySpan(null);
+    }
+
+    @Override
+    public OpenTelemetrySpan nextSpan(TraceContext traceContext) {
+      if (!(traceContext instanceof OpenTelemetryTraceContext)) {
+        return nextSpan();
+      }
+
+      Context context = ((OpenTelemetryTraceContext) traceContext).getContext();
+
+      final io.opentelemetry.trace.Span parent = TracingContextUtils.getSpanWithoutDefault(context);
+
+      return new OpenTelemetrySpan(parent);
+    }
+  }
+
+  // The order that callbacks will be called in or which thread they are called from is not well
+  // defined. We go ahead and buffer all data until we know we have a span. This implementation is
+  // particularly safe, synchronizing all accesses. Relying on implementation details would allow
+  // reducing synchronization but the impact should be minimal.
+  static class OpenTelemetrySpan extends Tracer.Span {
+    @Nullable private final Span parent;
+
+    @Nullable private OpenTelemetryEndpoint endpoint;
+    @Nullable private String name;
+
+    @Nullable private List<Object> events;
+    @Nullable private List<String> tags;
+
+    @Nullable private Status status;
+
+    @Nullable private Span span;
+
+    protected OpenTelemetrySpan(@Nullable Span parent) {
+      this.parent = parent;
+    }
+
+    // Called before start. We need to buffer because this until then.
+    @Override
+    public synchronized Tracer.Span name(String name) {
+      if (span != null) {
+        span.updateName(name);
+      } else {
+        this.name = name;
+      }
+
+      return this;
+    }
+
+    @Override
+    public synchronized Tracer.Span remoteEndpoint(Endpoint endpoint) {
+      if (endpoint instanceof OpenTelemetryEndpoint) {
+        if (span != null) {
+          fillEndpoint(span, (OpenTelemetryEndpoint) endpoint);
+        } else {
+          this.endpoint = (OpenTelemetryEndpoint) endpoint;
+        }
+      }
+      return this;
+    }
+
+    @Override
+    public synchronized Tracer.Span start() {
+      // If name() wasn't called yet we will update it later.
+      String spanName = name != null ? name : "REDIS";
+      Span.Builder builder = TRACER.spanBuilder(spanName).setSpanKind(Kind.CLIENT);
+      if (parent != null) {
+        builder.setParent(parent);
+      } else {
+        builder.setNoParent();
```

Comment:
If we don't call this, then we _may_ get a span from current Otel context, don't you think?